cppcache/src/ThinClientPoolDM.cpp

```diff
@@ -408,93 +408,90 @@ void ThinClientPoolDM::cleanStaleConnections(std::atomic<bool>& isRunning) {
 
   auto _idle = getIdleTimeout();
   auto _nextIdle = _idle;
-  {
-    TcrConnection* conn = nullptr;
 
-    std::vector<TcrConnection*> savelist;
-    std::vector<TcrConnection*> replacelist;
-    std::set<ServerLocation> excludeServers;
+  TcrConnection* conn = nullptr;
 
-    while ((conn = getNoWait()) != nullptr && isRunning) {
-      if (canItBeDeleted(conn)) {
-        replacelist.push_back(conn);
-      } else if (conn) {
-        auto nextIdle =
-            _idle - std::chrono::duration_cast<std::chrono::milliseconds>(
-                        TcrConnection::clock::now() - conn->getLastAccessed());
-        if (nextIdle > std::chrono::seconds::zero() && nextIdle < _nextIdle) {
-          _nextIdle = nextIdle;
-        }
-        savelist.push_back(conn);
+  std::vector<TcrConnection*> savelist;
+  std::vector<TcrConnection*> removelist;
+  std::set<ServerLocation> excludeServers;
+
+  while ((conn = getNoWait()) != nullptr && isRunning) {
+    if (canItBeDeleted(conn)) {
+      removelist.push_back(conn);
+    } else if (conn) {
+      auto nextIdle =
+          _idle - std::chrono::duration_cast<std::chrono::milliseconds>(
+                      TcrConnection::clock::now() - conn->getLastAccessed());
+      if (nextIdle > std::chrono::seconds::zero() && nextIdle < _nextIdle) {
+        _nextIdle = nextIdle;
       }
+      savelist.push_back(conn);
     }
+  }
 
-    size_t replaceCount =
-        m_attrs->getMinConnections() - static_cast<int>(savelist.size());
+  auto replaceCount =
+      m_attrs->getMinConnections() - static_cast<int>(savelist.size());
 
-    LOGDEBUG("Preserving %d connections", savelist.size());
+  LOGDEBUG("Preserving %d connections", savelist.size());
 
-    for (auto savedconn : savelist) {
-      put(savedconn, false);
-    }
-    savelist.clear();
-    int count = 0;
+  for (auto savedconn : savelist) {
+    put(savedconn, false);
+  }
+  savelist.clear();
+  int count = 0;
 
-    for (std::vector<TcrConnection*>::const_iterator iter = replacelist.begin();
-         iter != replacelist.end(); ++iter) {
-      TcrConnection* conn = *iter;
-      if (replaceCount <= 0) {
-        GF_SAFE_DELETE_CON(conn);
-        removeEPConnections(1, false);
-        getStats().incLoadCondDisconnects();
-        LOGDEBUG("Removed a connection");
+  for (std::vector<TcrConnection*>::const_iterator iter = removelist.begin();
+       iter != removelist.end(); ++iter) {
+    TcrConnection* conn = *iter;
+    if (replaceCount <= 0) {
+      GF_SAFE_DELETE_CON(conn);
```

Comment:
For another PR, sure would be nice to replace all these raw pointers with `std::shared_ptr` or `std::unique_ptr`. Look at all these places we delete, and potentially leak, connections.